src/server/cluster/outgoing_slot_migration.cc

```diff
@@ -81,7 +81,8 @@ class OutgoingMigration::SliceSlotMigration : private ProtocolClient {
   }
 
   void Cancel() {
-    cntx_.Cancel();
+    // Close socket for clean disconnect.
+    CloseSocket();
```

Comment:
I think `streamer_.Cancel();` should work as is.

Reply:
I found that we do the same for replica. And streamer_.Cancel(); doesn't work without close socket because we need to cancel OnCompletion callback

Reply:
Any idea why we are not WaitForInflightToComplete inside JournalStreamer::Cancel if cntx is cancelled?

`void JournalStreamer::Cancel() {
    VLOG(1) << "JournalStreamer::Cancel";
    waker_.notifyAll();
    journal_->UnregisterOnChange(journal_cb_id_);
    if (!cntx_->IsCancelled()) {
      WaitForInflightToComplete();
    }
}`

Reply:
Because we can have different situations
1) we have an error and cancel by error
2) we cancel for our reason
3) everything works good
4) was canceled by an error in another fiber

These different scenarios have different behavior and  WaitForInflightToComplete can wait up to 2 minutes (connection timeout) without any result so if we cancel the connection by our reason there is no sense to wait so long period of time

Reply:
From my point of view we should refactor JournalStreamer, but such refactoring affect replication and I don't think that we should do it right now